File: src/models/StoryFilter.ts
import {Entity,PrimaryGeneratedColumn,Column,ManyToOne,JoinColumn,} from 'typeorm';
import { Story } from './Story';

@Entity('story_filter')
export class StoryFilter {
  @PrimaryGeneratedColumn()
  id!: number;

  @Column()
  story_id!: number;

  @Column({ type: 'varchar' })
  type!: string; // personagens, generos, tags

  @Column({ type: 'varchar', nullable: true })
  description!: string;

  @ManyToOne(() => Story, (story) => story.filters)
  @JoinColumn({ name: 'story_id' })
  story!: Story;
}

File: backend/src/models/story_filter.py
from sqlalchemy import Column, Integer, String, ForeignKey
from sqlalchemy.orm import relationship
from src.db.database import Base


class StoryFilter(Base):
    __tablename__ = "story_filter"

    id = Column(Integer, primary_key=True, index=True)
    story_id = Column(Integer, ForeignKey("story.id"), nullable=False)
    type = Column(String, nullable=False)  # personagens, generos, tags
    description = Column(String, nullable=True)

    story = relationship("Story", back_populates="filters")
